src/integer_mod_q/mat_zq/get.rs

```diff
@@ -0,0 +1,275 @@
+//! Implementations to get entries from a [`MatZq`] matrix.
+
+use super::MatZq;
+use crate::integer::Z;
+use crate::integer_mod_q::Modulus;
+use crate::traits::{GetEntry, GetNumColumns, GetNumRows};
+use crate::utils::coordinate::evaluate_coordinates;
+use crate::{error::MathError, integer_mod_q::Zq};
+use flint_sys::fmpz_mod_mat::fmpz_mod_mat_entry;
+use std::fmt::Display;
+
+impl MatZq {
+    /// Returns the modulus of the matrix as a [`Modulus`] value
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    ///
+    /// let matrix = MatZq::new(5, 10, 7).unwrap();
+    /// let entry = matrix.get_mod().unwrap();
+    /// ```
+    pub fn get_mod(&self) -> Result<Modulus, MathError> {
+        Modulus::try_from_z(&Z {
+            value: self.matrix.mod_[0],
+        })
+    }
```

Comment:
This creates a memory leak -> create Z with pointer -> drop Z -> content behind pointer dropped -> memory leak
```suggestion
        let mut out_z = Z::default();
        unsafe{ fmpz_set(&mut out.value, &self.matrix.mod_[0])};
        Modulus::try_from_z(out_z).unwrap()
    }
```

Reply:
I would not unwrap here, but use an "expect" instead.